auth-tokens/src/main/java/com/palantir/tokens/auth/BearerToken.java

```diff
@@ -32,6 +40,47 @@
     private static final String VALIDATION_PATTERN_STRING = "^[A-Za-z0-9\\-\\._~\\+/]+=*$";
     private static final Pattern VALIDATION_PATTERN = Pattern.compile(VALIDATION_PATTERN_STRING);
 
+    private static final String SUBJECT_FIELD = "sub";
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    /**
+     * If the token contains a 'sub' field, as a base64 encoded UUID, return it as a String.
+     * @return the 'sub' field of the JWT, if it exists
+     */
+    public final Optional<String> getUserIdInsecure() {
```

Comment:
I'm in sync that this library can and should likely be Palantir specific, but I don't think the functionality encoded here belongs in this method on this class.

We likely want to parse out other JWT content, such as the sessionId, and also the expiration time, and it seems reasonable that in keeping with this project's semantic and narrow definitions to add a method like `JsonWebToken asJwt()` to this class, then to stick parsing of `userId`, `sessionId` and `expiration` into `JsonWebToken`.

Your logging examples wouldn't be much more complex:

    logger.warn("User {} ...",
        bearerToken.asJwt().getUserId())

Separately, we should likely consider caching some state here so that we're not re-parsing the whole thing if we want to render multiple fields:

    logger.warn("User {} with Session {} ...",
        bearerToken.asJwt().getUserId(),
        bearerToken.asJwt().getSessionId())

As an alternative, if we give the JWT class a nice stringify/logger function, we can likely normalize how UserId and SessionId are written to the logs, which seems advantageous.